tiledb/sm/cpp_api/filter.h

```diff
@@ -343,6 +345,10 @@ class Filter {
         if (!std::is_same<uint8_t, T>::value)
           throw std::invalid_argument("Option value must be uint8_t.");
         break;
+      case TILEDB_TYPED_VIEW_OUTPUT_DATATYPE:
+        if (!std::is_same<tiledb_datatype_t, T>::value)
```

Comment:
This either needs to be `uint8_t` or the function signature needs specialization for `tiledb_datatype_t` since its not considered "numeric".